src/ja/user/message/add.py

```diff
@@ -33,6 +33,9 @@ def from_dict(cls, property_dict: Dict[str, object]) -> "AddCommand":
 
     def execute(self, database: ServerDatabase) -> Response:
         job: Job = self.config.job
+        if self.effective_user != 0 and self.effective_user != job.owner_id:
+            return Response(result_string="Cannot submit jobs for other users.", is_success=False)
```

Comment:
Shouldn't the command just set job.owner_id to self.effective_user?
I don't think Job.owner_id is ever changed from its default of -1 (outside of tests).

Reply:
wouldn't the command handler set the effective_user at line 80
`        user_command.effective_user = 0 if self._user_is_admin(user) else pwd.getpwnam(user).pw_uid`

Reply:
@JohannesGaessler  I do not think this is the case. My main consideration was that, at some point in the future, there might be a need for administrators to restart jobs or something like that. In which case, administrators would need to be able to submit jobs for other people.

Reply:
I think you are not understanding the problem:
The Job class has an owner_id property.
As of this PR commands also have an effective_user property.
For regular users permission checks are performed by comparing these two properties.
However, as far as I can tell we haven't yet implemented setting Job.owner_id.
Job.owner_id is initialized with -1 in the CLI handler and never changed from this default after that.
In any test that we did we always set Job.owner_id manually.
I think that if we do not change this regular users will not be able to add jobs at all.

If we want to allow admins to add jobs for other users I would suggest something like this:
If `Job.owner_id == -1` then `Job.owner_id = Command.effective_user` else `check_permissions()`.

Reply:
The proper fix would be to set the owner id in the user CLI, not here.

Reply:
No it wouldn't be.
The user client and the server are by design running on different systems.
Linux username and user IDs are generally **not** the same across systems.

Reply:
Didn't we say we have LDAP and the same users everywhere?

Reply:
The way I remember it is that we would have the same users on the server and the workers but **not** on the user clients.

Reply:
Okay, I had another impression since we were talking about mounting the user's home directory or something similar by default. This is not possible if the users are different, but whatever, should be fixed now.

Reply:
Well, if we were to mount the home directory of a user it would be the home directory of the user that is used for the SSH login, in other words effective_user, right?

Reply:
>This is not possible if the users are different

Why?

Reply:
Let's not waste time on this.

Reply:
Okay.